### PR TITLE
Bug: Fix NPE in time.Equal method

### PR DESCRIPTION
**Pull request by tamalsaha**

### CODE REVIEW

**Comment by dims** [comment]
@tamalsaha Can you please add a quick test too?

**Comment by tamalsaha** [comment]
@dims , added tests. PTAL.

**Comment by tamalsaha** [comment]
@dims , tests are passing.

**Comment by dims** [comment]
@tamalsaha Thanks. i've already approved. you will need lgtm/approval from those listed in the OWNERS file above

**Comment by tamalsaha** [comment]
@deads2k / @caesarxuchao may I get  a lgtm ?

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time.go [hunk 1]
```diff
@@ -80,7 +80,13 @@ func (t *Time) Before(u *Time) bool {
 
 // Equal reports whether the time instant t is equal to u.
 func (t *Time) Equal(u *Time) bool {
-	return t.Time.Equal(u.Time)
+	if t == nil && u == nil {
+		return true
+	} else if t != nil && u != nil {
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time.go]: golang and project style don't use an else

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time.go [hunk 2]
```diff
@@ -80,7 +80,13 @@ func (t *Time) Before(u *Time) bool {
...
+		return t.Time.Equal(u.Time)
+	} else {
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time.go]: else not not need

```go
if foo{
    return
}
if bar {
   return
}

return 
```

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time_test.go [hunk 3]
```diff
@@ -171,3 +171,24 @@ func TestTimeProto(t *testing.T) {
...
+		result bool
+	}{
+		{nil, nil, true},
+		{&t1, &t1, true},
+		{nil, &t1, false},
+		{&t1, nil, false},
+	}
+
+	for _, c := range cases {
+		result := c.x.Equal(c.y)
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time_test.go]: `t.Run(name, func(t *testing.T){})` please

**Comment by deads2k** [comment]
nits, squash and lgtm.

**Comment by tamalsaha** [comment]
@deads2k , I have addressed your comments. PTAL.

**Pull request merged**